Code/recipeApp.py

```diff
@@ -118,25 +122,41 @@ def user_input(self):
         self.user_ingredients = userinput.split() # splits each ingredient up
         # print("ingredients:", self.user_ingredients)
 
-    def compare_ingredients(self):
+    def compare_ingredients(self, w):
         """
         Compares the dataframe's ingredients list and the user's.
         This might be helpful: https://towardsdatascience.com/dealing-with-list-values-in-pandas-dataframes-a177e534f173
+        https://stackoverflow.com/questions/5319922/python-check-if-word-is-in-a-string
+
+        https://www.guru99.com/python-regular-expressions-complete-tutorial.html
         :return:
         """
 
+        return re.compile(r'\b({0})\b'.format(w), flags=re.IGNORECASE).search
```

Comment:
It looks like you're using regex to compare ingredient list, but that might be too hard? I think you could use regex to select the recipe ingredient out of each recipe phrase (i.e. strawberry out of 2 cups of strawberry), then use String comparison (recipe_ingredient in  user_ingredients) to see if the user has it

Reply:
^^ I agree. It makes more sense to get rid of the other things in the recipe and just leave the ingredients for comparison.